src/ResourceManager/LogicApp/Commands.LogicApp/Cmdlets/IntegrationAccount/GetAzureIntegrationAccountGeneratedX12IcnCommand.cs

```diff
@@ -0,0 +1,75 @@
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.LogicApp.Cmdlets
+{
+    using System.Collections.Generic;
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.LogicApp.Utilities;
+
+    /// <summary>
+    /// Gets the integration account generated X12 interchange control number by agreement name
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "AzureRmIntegrationAccountGeneratedX12Icn")]
+    [OutputType(typeof(IntegrationAccountControlNumber), typeof(IList<QualifiedIntegrationAccountControlNumber>))]
+    public class GetAzureIntegrationAccountGeneratedX12IcnCommand : LogicAppBaseCmdlet
+    {
+        #region Input Parameters
+
+        /// <summary>
+        /// Gets or sets the resource group name.
+        /// </summary>
+        [Parameter(Mandatory = true, HelpMessage = "The integration account resource group name.",
+            ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the integration account name.
+        /// </summary>
+        [Parameter(Mandatory = true, HelpMessage = "The integration account name.")]
+        [Alias("ResourceName")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets the agreement name.
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = "The integration account agreement name.")]
+        [ValidateNotNullOrEmpty]
+        public string AgreementName { get; set; }
+
+        #endregion Input Parameters
+
+        /// <summary>
+        /// Executes the get integration account X12 control number command.
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+
+            this.WriteObject(
+                sendToPipeline: string.IsNullOrEmpty(this.AgreementName) ?
+                    this.IntegrationAccountClient.ListIntegrationAccountGeneratedIcns(
+                        resourceGroupName: this.ResourceGroupName,
+                        integrationAccountName: this.Name,
+                        agreementType: Management.Logic.Models.AgreementType.X12) as object :
```

Comment:
@maxshao96 it looks like this line is the only one that differs between this cmdlet and `Get-AzureRmIntegrationAccountGeneratedEdifactIcn` (this cmdlet and `Get-AzureRmIntegrationAccountGeneratedIcn` are actually identical, so I don't know why we need a separate cmdlet)

Would it make sense to move the functionality of this cmdlet and `Get-AzureRmIntegrationAccountGeneratedEdifactIcn` into the existing `Get-AzureRmIntegrationAccountGeneratedIcn` cmdlet by adding SwitchParameters to allow users to toggle between the `agreementType` they want to use?

For example, this would enable three calls:
- `Get-AzureRmIntegrationAccountGeneratedIcn`
- `Get-AzureRmIntegrationAccountGeneratedIcn -Edifact`

(you can choose whatever names you like for these switches)

This allows us to only need one cmdlet to do what three cmdlets would do, all just by using different switches to jump into different parameter sets.